source/tls_channel_handler.c

```diff
@@ -62,82 +62,49 @@ void aws_tls_ctx_options_clean_up(struct aws_tls_ctx_options *options) {
     AWS_ZERO_STRUCT(*options);
 }
 
-static int s_load_null_terminated_buffer_from_cursor(
-    struct aws_byte_buf *load_into,
-    struct aws_allocator *allocator,
-    const struct aws_byte_cursor *from) {
-    if (from->ptr[from->len - 1] == 0) {
-        if (aws_byte_buf_init_copy_from_cursor(load_into, allocator, *from)) {
-            return AWS_OP_ERR;
-        }
-
-        load_into->len -= 1;
-    } else {
-        if (aws_byte_buf_init(load_into, allocator, from->len + 1)) {
-            return AWS_OP_ERR;
-        }
-
-        memcpy(load_into->buffer, from->ptr, from->len);
-        load_into->buffer[from->len] = 0;
-        load_into->len = from->len;
-    }
-
-    return AWS_OP_SUCCESS;
-}
-
-static int s_tls_ctx_options_pem_sanitize(struct aws_tls_ctx_options *options) {
-    (void)options;
-    /* PEM is not supported on iOS */
-#if !defined(AWS_OS_IOS)
-    if (!options) {
-        return AWS_OP_SUCCESS;
-    }
-    if (options->allocator) {
-        if (aws_sanitize_pem(&options->ca_file, options->allocator) |
-            aws_sanitize_pem(&options->certificate, options->allocator) |
-            aws_sanitize_pem(&options->private_key, options->allocator)) {
-            return AWS_OP_ERR;
```

Comment:
Instead of re-sanitizing N pem files whenever 1 of them is set, I just made sure to aws_sanitize_pem() each of these files when we set them.